providers/implementations/encode_decode/decode_der2key.c

```diff
@@ -304,8 +306,7 @@ static int der2key_export_object(void *vctx,
         /* The contents of the reference is the address to our object */
         keydata = *(void **)reference;
 
-        return export(keydata, OSSL_KEYMGMT_SELECT_ALL,
-                      export_cb, export_cbarg);
+        return export(keydata, ctx->selection, export_cb, export_cbarg);
```

Comment:
For the case where ctx->selection is 0 is that what we want to pass to the export?

Reply:
That's a matter of interpretation. Is that something that should be decided here or in the export function we're calling? I have opted for the latter

Reply:
So you have changed dh_export() to handle passing 0 as the selection for example?